test/integration/etcd/data.go

```diff
@@ -544,13 +544,20 @@ func GetEtcdStorageDataForNamespace(namespace string) map[schema.GroupVersionRes
 		// --
 	}
 
-	// k8s.io/kubernetes/pkg/apis/storage/v1beta1
 	// add csinodes if CSINodeInfo feature gate is enabled
 	if utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo) {
+		// k8s.io/kubernetes/pkg/apis/storage/v1beta1
 		etcdStorageData[gvr("storage.k8s.io", "v1beta1", "csinodes")] = StorageData{
 			Stub:             `{"metadata": {"name": "csini1"}, "spec": {"drivers": [{"name": "test-driver", "nodeID": "localhost", "topologyKeys": ["company.com/zone1", "company.com/zone2"]}]}}`,
 			ExpectedEtcdPath: "/registry/csinodes/csini1",
 		}
+
+		// k8s.io/kubernetes/pkg/apis/storage/v1
+		etcdStorageData[gvr("storage.k8s.io", "v1", "csinodes")] = StorageData{
+			Stub:             `{"metadata": {"name": "csini2"}, "spec": {"drivers": [{"name": "test-driver", "nodeID": "localhost", "topologyKeys": ["company.com/zone1", "company.com/zone2"]}]}}`,
+			ExpectedEtcdPath: "/registry/csinodes/csini2",
+			ExpectedGVK:      gvkP("storage.k8s.io", "v1beta1", "CSINode"),
```

Comment:
Is this right? I'm not actually sure where this object-specific ordering is specified. Since the storage/v1 group already exists, all the places I found where you specify ordering already prioritize the v1 group over v1beta1. Is [this](https://github.com/kubernetes/kubernetes/blob/14344b57e56258e87cbe80c8cd80399855eca424/pkg/kubeapiserver/default_storage_factory_builder.go#L60) relevant?

Reply:
I checked with liggitt. He clarified that there's a group default, which is set to v1, and then a per-type override, which is currently v1beta1.  For now, we keep the type override at v1beta1 for one more release, and then can remove it to use v1 in 1.18: https://github.com/kubernetes/kubernetes/issues/83513